projects/components/src/table/controls/table-controls-api.ts

```diff
@@ -30,26 +30,29 @@ export const enum TableControlOptionType {
   UnsetFilter = 'unset-filter'
 }
 
-export interface TableControlOption<TMetaValue = unknown, TValue = unknown> {
-  type: TableControlOptionType;
-  label: string;
-  metaValue: TMetaValue; // Used in a query - type based on TableControlOptionType
-  value?: TValue; // If a control needs to carry a value, use this (example: checkbox boolean)
-}
+export type TableControlOption<T = unknown> =
+  | TableUnsetFilterControlOption<T>
+  | TableFilterControlOption<T>
+  | TablePropertyControlOption<T>;
 
-export interface TableUnsetFilterControlOption extends TableControlOption<string, undefined> {
+interface TableControlOptionBase<T> {
```

Comment:
Pfft. Who uses "Base" in a name? 🙄

Reply:
Hah - I hated every minute of it, but `TableControlOption` was already taken by someone!